src/providers/mod.rs

```diff
@@ -73,13 +91,13 @@ fn write_ssh_keys(user: User, ssh_keys: Vec<PublicKey>) -> Result<()> {
     if !ssh_keys.is_empty() {
         // open the user's authorized keys directory
         let user_name = user.name().to_string_lossy().into_owned();
+        let path = AuthorizedKeys.ssh_dir.display();
```

Comment:
This should be `let path = authorized_keys_dir.ssh_dir.display();`. So then you'd put it lower down. Could just be right before the call to `write_ssh_key_journal_entry`.